src/Type/Php/CurlInitReturnTypeExtension.php

```diff
@@ -26,10 +41,49 @@ public function getTypeFromFunctionCall(
 		Scope $scope,
 	): Type
 	{
-		$argsCount = count($functionCall->getArgs());
+		$args = $functionCall->getArgs();
+		$argsCount = count($args);
 		$returnType = ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
+		$notFalseReturnType = TypeCombinator::remove($returnType, new ConstantBooleanType(false));
 		if ($argsCount === 0) {
-			return TypeCombinator::remove($returnType, new ConstantBooleanType(false));
+			return $notFalseReturnType;
+		}
+
+		$urlArgType = $scope->getType($args[0]->value);
+		if ($urlArgType->isNull()->yes()) {
+			return $notFalseReturnType;
+		}
+		if ($urlArgType->isString()->yes()) {
+			$urlArgValue = $urlArgType->getConstantScalarValues()[0] ?? null;
+			if ($urlArgValue !== null) {
+				if (!is_string($urlArgValue)) {
+					throw new ShouldNotHappenException();
+				}
```

Comment:
This feels a bit cumbersome to do, but I couldn't find a simpler way to get the constant value of a string.
Maybe a `Type::getConstantStringValue(): ?string` would be useful. But then that might lead to `Type` exploding with helper functions eventually.

Reply:
Does `Type::getConstantStrings` work for you?

Reply:
Ah I missed that method. But for multiple constant values it turned out to not be helpful. I overlooked that possibility completely.
Thanks a lot!